Members: allow user to choose what to sort by

vladh opened the pull request:
Staging deployment: https://osp-staging.vlad.website/members/

chadwhitacre reviewed (commented):
Ooh! I was thinking we would make it optional, if I'm reading this right it changes the sort from one to the other but the viewer can't chose, correct?

vladh commented:
@chadwhitacre Got it! This is now ready for review. The user can now select what to sort by. 🎉 

Staging deployment: https://osp-staging.vlad.website/members/

## Desktop

![2025-05-16-09-42-39](https://github.com/user-attachments/assets/a973999a-2e91-4248-82cf-85d941117920)
![2025-05-16-09-42-46](https://github.com/user-attachments/assets/56eb0524-f757-44d7-9a00-46c7efee0581)
![2025-05-16-09-42-52](https://github.com/user-attachments/assets/f3d375e2-e26c-4113-a787-45ba50106a9c)
![2025-05-16-09-42-58](https://github.com/user-attachments/assets/942aede7-c7a6-4ca9-95cc-923d8a04899f)

## Mobile

![IMG_9517](https://github.com/user-attachments/assets/fc4a66b6-38f3-460e-ac6f-663d8e9861f0)
![IMG_9518](https://github.com/user-attachments/assets/40ff3e22-f923-475c-b269-bafdbd9e3d79)
![IMG_9519](https://github.com/user-attachments/assets/35c80712-b4f8-4240-b13b-1edd3a65205f)

chadwhitacre reviewed (approved):
Yay! Should we add the date itself in the UI?

chadwhitacre commented:
Another follow on would be asc/desc

vladh commented:
I'm afraid that adding the date to the cards will make them too cluttered…but I can try it!

Can absolutely add asc/desc to the sort if you think it's useful.

vladh commented:
@chadwhitacre I've added ascending and descending sorting:

![2025-05-16-14-20-44](https://github.com/user-attachments/assets/665a1e07-9d00-4ce3-98d3-e4c1bf0bea78)

I have not added the member join date to the UI of the member list. Adding it would make the card view cluttered, and it just won't fit in the table, especially on mobile.

Instead, I've added the join date to the individual member pages:

![2025-05-16-14-23-27](https://github.com/user-attachments/assets/43fdbb23-6f7d-494c-b94c-2e8cbd21dcc0)

What do you think?

Happy to do any tweaks you'd like, otherwise this is good to merge whenever.

vladh commented:
https://github.com/user-attachments/assets/c40cc87f-1afc-43ea-8d63-a2f1d6bf8063
